tests/integration/components/container-query/dataAttributePrefix-test.js

```diff
@@ -304,4 +304,151 @@ module('Integration | Component | container-query', function(hooks) {
       });
     });
   });
+
+
+  module('When @dataAttributePrefix is updated', function(hooks) {
+    hooks.beforeEach(async function() {
+      this.dataAttributePrefix = 'cq1';
+
+      await render(hbs`
+        <div
+          data-test-parent-element
+          style="width: 250px; height: 500px;"
+        >
+          <ContainerQuery
+            @features={{hash
+              small=(cq-width max=300)
+              medium=(cq-width min=300 max=600)
+              large=(cq-width min=600 max=900)
+              short=(cq-height max=500)
+              tall=(cq-height min=500)
+              ratio-type-A=(cq-aspect-ratio min=0.25 max=0.75)
+              ratio-type-B=(cq-aspect-ratio min=0.5 max=1.5)
+              ratio-type-C=(cq-aspect-ratio min=1.25 max=2)
+            }}
+            @dataAttributePrefix={{this.dataAttributePrefix}}
+            as |CQ|
+          >
+            <p data-test-feature="small">{{CQ.features.small}}</p>
+            <p data-test-feature="medium">{{CQ.features.medium}}</p>
+            <p data-test-feature="large">{{CQ.features.large}}</p>
+
+            <p data-test-feature="short">{{CQ.features.short}}</p>
+            <p data-test-feature="tall">{{CQ.features.tall}}</p>
+
+            <p data-test-feature="ratio-type-A">{{CQ.features.ratio-type-A}}</p>
+            <p data-test-feature="ratio-type-B">{{CQ.features.ratio-type-B}}</p>
+            <p data-test-feature="ratio-type-C">{{CQ.features.ratio-type-C}}</p>
+
+            <p data-test-width-height>{{CQ.dimensions.width}} x {{CQ.dimensions.height}}</p>
+            <p data-test-aspect-ratio>{{CQ.dimensions.aspectRatio}}</p>
+          </ContainerQuery>
+        </div>
+      `);
+
+      this.set('dataAttributePrefix', 'cq2');
+    });
+
+
+    test('The component doesn\'t update the data attributes', async function(assert) {
+      assert.areDataAttributesCorrect({
+        'data-cq1-small': '',
+        'data-cq1-medium': undefined,
+        'data-cq1-large': undefined,
+        'data-cq1-short': undefined,
+        'data-cq1-tall': '',
+        'data-cq1-ratio-type-A': '',
+        'data-cq1-ratio-type-B': '',
+        'data-cq1-ratio-type-C': undefined
+      });
+
+      assert.areDataAttributesCorrect({
+        'data-cq2-small': undefined,
+        'data-cq2-medium': undefined,
+        'data-cq2-large': undefined,
+        'data-cq2-short': undefined,
+        'data-cq2-tall': undefined,
+        'data-cq2-ratio-type-A': undefined,
+        'data-cq2-ratio-type-B': undefined,
+        'data-cq2-ratio-type-C': undefined
+      });
```

Comment:
Ideally, all `data-cq1-*` attributes would map to `undefined`, while all `data-cq2-*` attributes to their correct values.